tensorflow_probability/examples/cifar10_bnn.py

```diff
@@ -0,0 +1,246 @@
+
+The architecture is LeNet-5 [1].
+
+#### References
+
+[1]: Yann LeCun, Leon Bottou, Yoshua Bengio, and Patrick Haffner.
+     Gradient-based learning applied to document recognition.
+     _Proceedings of the IEEE_, 1998.
+     http://yann.lecun.com/exdb/publis/pdf/lecun-01a.pdf
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+# Dependency imports
+from absl import flags
+import matplotlib
+matplotlib.use("Agg")
+#from matplotlib import figure  # pylint: disable=g-import-not-at-top
+#from matplotlib.backends import backend_agg
+import numpy as np
+import tensorflow as tf
+#import tensorflow_probability as tfp
+
+from tensorflow.python.keras.datasets import cifar10
+
+import seaborn as sns  # pylint: disable=g-import-not-at-top
+from models.BResNet import BayesResNet
+from models.BVGG import BVGG
+
+# TODO(b/78137893): Integration tests currently fail with seaborn imports.
+import warnings
+warnings.simplefilter(action='ignore')
+
+#try:
+#  import seaborn as sns  # pylint: disable=g-import-not-at-top
+#  HAS_SEABORN = True
+#except ImportError:
+#  HAS_SEABORN = False
+
+tfd = tf.contrib.distributions
+
+IMAGE_SHAPE = [32, 32, 3]
+
+flags.DEFINE_float("learning_rate",
+                   default=0.0001,
+                   help="Initial learning rate.")
+flags.DEFINE_integer("epochs",
+                     default=700,
+                     help="Number of epochs to train for.")
+flags.DEFINE_integer("batch_size",
+                     default=128,
+                     help="Batch size.")
+flags.DEFINE_string("data_dir",
+                    default=os.path.join(os.getenv("TEST_TMPDIR", "/tmp"),
+                                         "bayesian_neural_network/data"),
+                    help="Directory where data is stored (if using real data).")
+flags.DEFINE_string(
+    "model_dir",
+    default=os.path.join(os.getenv("TEST_TMPDIR", "/tmp"),
+                         "bayesian_neural_network/"),
+    help="Directory to put the model's fit.")
+flags.DEFINE_integer("eval_freq",
+                     default=400,
+                     help="Frequency at which to validate the model.")
+flags.DEFINE_integer("num_monte_carlo",
+                     default=50,
+                     help="Network draws to compute predictive probabilities.")
+flags.DEFINE_string("arch",
+                  default="vgg",
+                  help="Network architecture to use.")
+flags.DEFINE_integer("prior_qmean",
+                     default=-9,
+                     help="Initial posterior mean of the log variance for q(w)")
+flags.DEFINE_float("variance_threshold",
+                     default=0.2,
+                     help="Posterior variance threshold constraint. Log var <= log(threshold).")
+flags.DEFINE_integer("kl_annealing",
+                     default=100,
+                     help="Epochs to anneal the KL term (anneals from 0 to 1)")
+
+FLAGS = flags.FLAGS
+
```

Comment:
You should run `pep8`. For example, you need two blanklines before after every top-level function def.